avr/cores/megacommand/MCL/SeqPage.h

```diff
@@ -27,20 +27,68 @@ class SeqPage : public LightPage {
   static uint8_t ignore_button_release;
   static bool show_track_menu;
 
+  bool recording = false;
+  char info1[8] = { '\0' };
+  char info2[8] = { '\0' };
+  uint8_t timeout_values[4] = { 0 }; // 255 == highlight
+
   SeqPageMidiEvents seqpage_midi_events;
   SeqPage(Encoder *e1 = NULL, Encoder *e2 = NULL, Encoder *e3 = NULL,
           Encoder *e4 = NULL)
       : LightPage(e1, e2, e3, e4) {
   }
-  virtual bool handleEvent(gui_event_t *event);
   void create_chars_seq();
   void draw_lock_mask(uint8_t offset, bool show_current_step = true);
   void draw_pattern_mask(uint8_t offset, uint8_t device, bool show_current_step = true);
-  void loop();
-  void display();
-  void setup();
-  void init();
-  void cleanup();
+  void draw_knob_frame();
+  void draw_knob(uint8_t i, const char* title, const char* text);
+  void draw_knob(uint8_t i, Encoder* enc, const char* name);
+  void select_track(uint8_t device, uint8_t track);
+
+  virtual bool handleEvent(gui_event_t *event);
+  virtual void loop();
+  virtual void display();
+  virtual void setup();
+  virtual void init();
+  virtual void cleanup();
+
+  static constexpr uint8_t pane_x1 = 0;
```

Comment:
These constant variables are going to be stored in RAM?

Might be better to change them to macro defined constants.

Reply:
These are const expressions and therefore inlined whenever used.
Think about them as the C++ answer to macros.